python/cudf/cudf/core/column/column.py

```diff
@@ -2193,166 +2208,72 @@ def as_column(
                 return cudf.core.column.ListColumn.from_sequences(arbitrary)
             raise
         return as_column(data, nan_as_null=nan_as_null)
-    else:
-        pa_type = None
+    elif not isinstance(arbitrary, (abc.Iterable, abc.Sequence)):
+        # TODO: This validation should probably be done earlier?
+        raise TypeError(
+            f"{type(arbitrary).__name__} must be an iterable or sequence."
+        )
+    from_pandas = nan_as_null is None or nan_as_null
+    if dtype is not None:
+        dtype = cudf.dtype(dtype)
         try:
-            if dtype is not None:
```

Comment:
I guess this whole block now gets handled by pandas or arrow for us?

Reply:
Correct (in the `elif`'s above). Mainly if a `dtype` is specified there's shouldn't be any `try/except` fallback